src/main/java/alfio/repository/TicketCategoryRepository.java

```diff
@@ -85,6 +85,15 @@
     @Query("select * from ticket_category_with_currency where event_id = :eventId  and tc_status = 'ACTIVE' order by ordinal asc, inception asc, expiration asc, id asc")
     List<TicketCategory> findAllTicketCategories(@Bind("eventId") int eventId);
 
+    @Query("select tc.* from ticket_category_with_currency tc" +
+        "    join ticket_category_statistics tcs on tc.id = tcs.ticket_category_id" +
+        "    where tc.event_id = :eventId" +
+        "    and tcs.is_expired is FALSE" +
+        "    and tcs.access_restricted is FALSE" +
+        "    and (tcs.bounded is FALSE or tcs.not_sold_tickets > 0)" +
+        " limit 1")
+    Optional<TicketCategory> findFirstWithAvailableTickets(@Bind("eventId") int eventId);
```

Comment:
Even if arbitrary, I would add a sort criteria, so it would consume the same category.